From b659e89f4479bc50afbadb45db0562abf501faa6 Mon Sep 17 00:00:00 2001
From: Li Jun <jun.li@nxp.com>
Date: Fri, 9 Jun 2017 18:38:06 +0800
Subject: [PATCH 1341/5242] MLK-15032-3 usb: chipidea: core: remove the extcon
 change for imx_4.1.y

commit  0d8b0537fd4eccc7089b4461b2f1383c605153da from
https://source.codeaurora.org/external/imx/linux-imx.git

commit 916e43e1d6fb ("MLK-13570-3 usb: chipidea: core: change extcon
usage for imx_4.1.y") is directly cherry-picked from 4.1.y, but which
is not valid anymore on 4.y kernel, so revert most part and only keep
the irq check after resume.

Reviewed-by: Peter Chen <peter.chen@nxp.com>
Signed-off-by: Li Jun <jun.li@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/usb/chipidea/core.c |   17 +++--------------
 1 file changed, 3 insertions(+), 14 deletions(-)

diff --git a/drivers/usb/chipidea/core.c b/drivers/usb/chipidea/core.c
index 24a70f3..9eb7a25 100644
--- a/drivers/usb/chipidea/core.c
+++ b/drivers/usb/chipidea/core.c
@@ -590,22 +590,11 @@ static int ci_cable_notifier(struct notifier_block *nb, unsigned long event,
 {
 	struct ci_hdrc_cable *cbl = container_of(nb, struct ci_hdrc_cable, nb);
 	struct ci_hdrc *ci = cbl->ci;
-	struct extcon_dev *dev = ptr;
-	int ret;
-
-	/* Only support ID extcon now */
 
-	ret = extcon_get_state(dev, EXTCON_USB_HOST);
-	if (ret && !cbl->connected) {
-		cbl->connected = true;
-		cbl->changed = true;
-		ci_irq(ci->irq, ci);
-	} else if (!ret && cbl->connected) {
-		cbl->connected = false;
-		cbl->changed = true;
-		ci_irq(ci->irq, ci);
-	}
+	cbl->connected = event;
+	cbl->changed = true;
 
+	ci_irq(ci->irq, ci);
 	return NOTIFY_DONE;
 }
 
-- 
1.7.9.5

